include/nbl/builtin/hlsl/complex.hlsl

```diff
@@ -17,19 +14,21 @@ namespace hlsl
 
 
 
-template <typename T>
-struct complex_t<vector<T, uint16_t N>>
+#ifdef __HLSL_VERSION
+
+
+template <typename T, uint16_t N>
+struct complex_t
 {
-    vector<T, N> real;
-    vector<T, N> imag;
+    T real[N];
+    T imag[N];
 };
```

Comment:
Why does complex_t need a dimension?

Reply:
it does not, the dimensionality is supposed to come from `T=vector<float_type,N`